src/pages/notifications/index.vue

```diff
@@ -0,0 +1,375 @@
+            isCivicLiker = this.getUserInfoByAddress(this.getAddress)
+              ?.isSubscribedCivicLiker;
+            i18nPath = 'event_list_page_event_message_purchase_nft';
+            fromName = this.$t('event_list_page_event_self');
+            toName = this.getUserInfoByAddress(creator)?.displayName || creator;
+            // eslint-disable-next-line prefer-destructuring
+            price = e.price;
+            memo = e.granterMemo || '';
+            break;
+
+          case 'nft_sale':
+            fromName =
+              this.getUserInfoByAddress(e.receiver)?.displayName || e.receiver;
+            // eslint-disable-next-line prefer-destructuring
+            price = e.price;
+            i18nPath = 'event_list_page_event_message_nft_sale';
+            displayAvatar = this.getUserInfoByAddress(e.receiver)?.avatar;
+            isCivicLiker = this.getUserInfoByAddress(e.receiver)
+              ?.isSubscribedCivicLiker;
+            memo = e.granterMemo || '';
+            break;
+
+          case 'receive_nft':
+            fromName =
+              this.getUserInfoByAddress(e.sender)?.displayName || e.sender;
+            i18nPath = 'event_list_page_event_message_receive_nft';
+            displayAvatar = this.getUserInfoByAddress(e.sender)?.avatar;
+            isCivicLiker = this.getUserInfoByAddress(e.sender)
+              ?.isSubscribedCivicLiker;
+            break;
+
+          case 'send_nft':
+            fromName = this.$t('event_list_page_event_self');
+            toName =
+              this.getUserInfoByAddress(e.receiver)?.displayName || e.receiver;
+            i18nPath = 'event_list_page_event_message_send_nft';
+            displayAvatar = this.getUserInfoByAddress(e.sender)?.avatar;
+            isCivicLiker = this.getUserInfoByAddress(e.sender)
+              ?.isSubscribedCivicLiker;
+            break;
+
+          default:
+            break;
+        }
+        if (memo) {
+          message = memo;
+        }
+
+        return {
+          displayAvatar,
+          isCivicLiker,
+          eventHasSeen,
+          message,
+          fromName,
+          toName,
+          price,
+          nftName,
+          i18nPath,
+          targetRoute: {
+            name: e.nft_id ? 'nft-class-classId-nftId' : 'nft-class-classId',
+            params: {
+              classId: e.class_id,
+              nftId: e.nft_id,
+            },
+          },
+          ...e,
+        };
+      });
+    },
+    groupedEventsByTime() {
+      return this.convertToGroupedEvents(this.processedEvents);
+    },
+  },
+  async mounted() {
+    if (this.getAddress && !this.walletHasLoggedIn) {
+      try {
+        await this.signLogin();
+      } catch {
+        // No-op
+      }
+    }
+    window.addEventListener('beforeunload', () => {
+      this.updateEventLastSeenTs();
+    });
+  },
+  beforeDestroy() {
+    window.removeEventListener('beforeunload', () => {
+      this.updateEventLastSeenTs();
+    });
```

Comment:
You need to use the same function for `addEventListener` and `removeEventListener`, in which `() => {}` is always a new function and not the same one.
Also does SPA navigation (i.e. clicking `nuxt-link` or `this.$router.push()`) trigger `beforeunload` ?